Fix warnings in Dunfell OE build.

shruthi-ravi opened the pull request:
There are still a few warnings left that I haven't figured out. Pushing what I have.
@ni/rtos

shruthi-ravi committed:
u-boot: Drop u-boot recipe bbappend in Dunfell.

NI ARM targets will not be supported in Dunfell. u-boot has updated
recipe files upstream and it will take significant effort to rework
these bbappends.
Drop all u-boot files and patches.

Signed-off-by: Shruthi Ravichandran <shruthi.ravichandran@ni.com>

shruthi-ravi committed:
python: Drop python2 bbappend recipe.

Python2 is no longer supported upstream.

Signed-off-by: Shruthi Ravichandran <shruthi.ravichandran@ni.com>

shruthi-ravi committed:
policycoreutils: Drop patch and bbappend recipe.

Patch has been merged upstream since v2.5.

Signed-off-by: Shruthi Ravichandran <shruthi.ravichandran@ni.com>

shruthi-ravi committed:
iproute2: Drop patches available upstream.

Signed-off-by: Shruthi Ravichandran <shruthi.ravichandran@ni.com>

shruthi-ravi committed:
gcc: Update Python bbappends indentation.

Address multiple warnings related to Python requiring 4 spaces
indentation instead of tabs.

Signed-off-by: Shruthi Ravichandran <shruthi.ravichandran@ni.com>

shruthi-ravi committed:
udev: Drop patch for wl12xx.

ARM targets not supported in Dunfell.

Signed-off-by: Shruthi Ravichandran <shruthi.ravichandran@ni.com>

shruthi-ravi committed:
libcap: Drop patches already merged upstream.

Signed-off-by: Shruthi Ravichandran <shruthi.ravichandran@ni.com>

shruthi-ravi committed:
docker: Drop bbappend recipe.

docker-registry not recommended anymore in OE recipe.

Signed-off-by: Shruthi Ravichandran <shruthi.ravichandran@ni.com>

shruthi-ravi committed:
libnss-mdns: Use wildcard in bbappend version.

Signed-off-by: Shruthi Ravichandran <shruthi.ravichandran@ni.com>

shruthi-ravi committed:
tzdata: Move bbappend to new recipe location.

Signed-off-by: Shruthi Ravichandran <shruthi.ravichandran@ni.com>

shruthi-ravi committed:
readline: Bump recipe bbappend version.

Signed-off-by: Shruthi Ravichandran <shruthi.ravichandran@ni.com>

shruthi-ravi committed:
opencv: Bump recipe bbappend version.

Signed-off-by: Shruthi Ravichandran <shruthi.ravichandran@ni.com>

shruthi-ravi committed:
coreutils: Fix warnings due to upstream updates.

Change priority of packages to a value higher than the priority of
coreutils. This fixes a warning that the binaries have multiple
providers with the same priority.
Update RDEPENDS variable name to fix warning that this key replaces
the original RDEPENDS_coreutils key in coreutils instead of
appending to it.

Signed-off-by: Shruthi Ravichandran <shruthi.ravichandran@ni.com>

amstewart commented on recipes-connectivity/openssl/openssl_1.0.2o.bbappend:
```diff
@@ -1,31 +0,0 @@
-FILESEXTRAPATHS_prepend := "${THISDIR}/files:"
-
-SRC_URI_remove = "file://debian/pic.patch"
```
Do you know the rationale for dropping this OE file? Is this a change we need to carry forward into a new openssl `.bbappend`?

shruthi-ravi replied:
Good question. The answer is likely? [This](https://github.com/ni/meta-nilrt/commit/46a320ade106e0744b1f42f244187448d1048188#diff-d12441f3749633acb88494958beb3909) is the commit that added the lines. [This](https://review-board.natinst.com/r/128139/) is the rb link describing the changes a little more. I guess NI-SSL is still a thing on Windows?

amstewart replied:
I think so. We should probably keep this part of the .bbappend.

shruthi-ravi replied:
Kept the following line
`SRC_URI_remove = "file://debian/pic.patch"`
This patch is not in our files directory. So we wouldn't need this definition right?
`FILESEXTRAPATHS_prepend := "${THISDIR}/files:"`

amstewart replied:
Correct; we don't need that `FILESEXTRAPATHS` assignment.

amstewart commented on recipes-devtools/python/python_%.bbappend:
```diff
@@ -1,5 +0,0 @@
-# we don't want python 2 to take precedence over the default python 3 because
-# it breaks salt if it gets installed from the feeds alongside python 3
-do_install_append () {
-	rm -f ${D}${bindir}/python
```
This is fine.

Have you verified that `/usr/bin/python` evaluates to python3 on the images? The python links in OE are kind of odd and I wouldn't be surprise if a relatively simple change like this messes something up.

shruthi-ravi replied:
Verified that /usr/bin/python evaluates to python3. Python 3.8.

amstewart commented on recipes-core/udev/udev/0001-ni-skip-kmod-for-wl12xx.patch:
```diff
@@ -4,7 +4,7 @@ index 8551f47..ac8bb60 100644
 +++ b/rules/80-drivers.rules
 @@ -2,6 +2,9 @@
  
- ACTION=="remove", GOTO="drivers_end"
+ ACTION!="add", GOTO="drivers_end"
  
 +# skip kmod for wl12xx device to prevent 30s delay issues with 3.2 kernel
 +ENV{MODALIAS}=="platform:wl12xx", GOTO="drivers_end"
```
I think this patch can be dropped, in its entirety, from the Dunfell release.

Also, you can close [this work item](https://dev.azure.com/ni/DevCentral/_workitems/edit/1066354) while you're at it. :)

shruthi-ravi replied:
Done.

amstewart commented on recipes-core/coreutils/coreutils_8.%.bbappend:
```diff
@@ -11,14 +11,16 @@ DEPENDS += "shadow-native pseudo-native niacctbase"
 RDEPENDS_${PN}-hostname += "niacctbase"
 
 # preserve existing behavior when installing base package
-RDEPENDS_${PN}_class-target += "\
```
I think the way you're supposed to fix this error is by using something like
```bash
RDEPENDS_${PN}_class-target_append += ...
```
... or something like that. I might be mistaken, but I would like to preserve the `${PN}` semantic, if we can.

shruthi-ravi replied:
Tried that, but it didn't work. Take a look at the final comment from Richard Purdie in this [thread](https://patchwork.openembedded.org/patch/98957/) that might explain why.

amstewart replied:
Ah; that's slightly annoying. Could you add a section to the bitbake wiki document with a link to that thread. I imagine this will come up many more times. :/

amstewart reviewed (changes requested):
Looks good overall. It's nice to see that we're able to drop so many `.patch`es!

shruthi-ravi committed:
openssl: Bump version and drop patches.

Bump version to match OE version 1.1.1g.
Drop patches for Pharlap and VxWorks. These OSes will not be
supported in Dunfell.
Drop patches for nibuild Windows. These patches have already been
applied in this version.

shruthi-ravi committed:
Drop GCCVERSION variable definition.

This variable gets defined in oe-core if not already defined. We
shouldn't have to maintain this version separately.

Signed-off-by: Shruthi Ravichandran <shruthi.ravichandran@ni.com>

shruthi-ravi committed:
libglade: move recipe from meta-openembedded.

Instead of carrying this patch forward every rebase in
meta-openembedded, move recipe to meta-nilrt.

commit 9dcf44f ("libglade: remove") dropped libglade because it
was broken by recipe specific sysroots (glib-2.0-native was missing).

Though it is old/deprecated/unmaintained, libglade is useful because
it's a dependency for the florence virtual keyboard which is the
smallest virtual keyboard in terms of disk space / memory usage.

Signed-off-by: Shruthi Ravichandran <shruthi.ravichandran@ni.com>

amstewart reviewed (changes requested):
> Though is old/deprecated/unmaintained

s/is/it is/
:)

amstewart commented on recipes-connectivity/openssl/openssl_1.1.1g.bbappend:
```diff
@@ -0,0 +1 @@
+SRC_URI_remove = "file://debian/pic.patch"
```
Could you add a comment in this `.bbappend` which explains why we're removing this `.patch`. i think that would be easier than having our future selves dig back through the git log to rediscover the rationale.

shruthi-ravi replied:
Done.

amstewart commented:
Looks good; pending the nits above.

shruthi-ravi commented:
I had updated the iproute2 version to 2.5 in another commit. I need to rebase this branch on top of those commits, which probably means another round of force pushes.
